MonoGame.Framework/Graphics/Effect/Effect.cs

```diff
@@ -69,9 +69,11 @@ protected Effect(Effect cloneSource)
             Clone(cloneSource);
 		}
 
-        public Effect (GraphicsDevice graphicsDevice, byte[] effectCode)
+        public Effect (GraphicsDevice graphicsDevice, byte[] effectCode, int index = 0, int count = -1)
```

Comment:
I was a little worried about this at first, but really it is one of the few places where we don't have a buffer range option.

I would just break it into two constructors:

```
public Effect (GraphicsDevice graphicsDevice, byte[] effectCode)
  : this(graphicsDevice, effectCode, 0, effectCode.Length)
{
}

public Effect (GraphicsDevice graphicsDevice, byte[] effectCode, int offset, int count)
: this(graphicsDevice)
{
```
```